lib/impure/db_sqlite.nim

```diff
@@ -716,8 +719,64 @@ proc setEncoding*(connection: DbConn, encoding: string): bool {.
 proc finalize*(sqlPrepared:SqlPrepared){.discardable.} = 
   discard finalize(sqlPrepared.PStmt)
 
+template dbBindParamError*(paramIdx: int,val:untyped) =
+  ## Raises a `DbError` exception.
+  var e: ref DbError
+  new(e)
+  e.msg = "error binding param in position " & $paramIdx 
+  raise e
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int32) =
+  ## Binds a int32  to the specified paramIndex.
+  if SQLITE_OK != bind_int(ps.PStmt, paramIdx.int32, val):
+    dbBindParamError(paramIdx,val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int) =
+  ## Binds a int32  to the specified paramIndex.
+  if SQLITE_OK != bind_int(ps.PStmt, paramIdx.int32, val.int32):
+    dbBindParamError(paramIdx,val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int,val: int64) =
+  ## Binds a int64  to the specified paramIndex.
+  if SQLITE_OK != bind_int64(ps.PStmt, paramIdx.int32, val):
+    dbBindParamError(paramIdx,val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: float64) =
+  ## Binds a 64bit float to the specified paramIndex.
+  if SQLITE_OK != bind_double(ps.PStmt, paramIdx.int32, val):
+    dbBindParamError(paramIdx,val)
+
+proc bindNull*(ps: SqlPrepared, paramIdx: int) =
+  ## Sets the bindparam at the specified paramIndex to null 
+  ## (default behaviour by sqlite).
+  if SQLITE_OK != bind_null(ps.PStmt, paramIdx.int32):
+    dbBindParamError(paramIdx,val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int,val: string) =
+  ## Binds a string to the specified paramIndex.
+  if SQLITE_OK != bind_text(ps.PStmt, paramIdx.int32,val.cstring,-1.int32 , SQLITE_STATIC):
+    dbBindParamError(paramIdx,val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int,val: cstring) =
+  ## binds a blob to the specified paramIndex.
```

Comment:
The typical Nim standard for a blob is a sequence of bytes or `openArray[byte]`. We never use `cstring` for these things.

If you want to provide a special bind for `cstring`, make it so that the user has to specify the size, since we don't know if the `cstring` buffer is a NUL-terminated one to pass to `len()`